tests/test_event_builder.py

```diff
@@ -139,6 +139,87 @@ def test_create_impression_event__with_attributes(self):
                                 event_builder.EventBuilder.HTTP_VERB,
                                 event_builder.EventBuilder.HTTP_HEADERS)
 
+  def test_create_impression_event_when_attribute_is_not_in_datafile(self):
+      """ Test that create_impression_event creates Event object
+      with right params when attribute is not in the datafile. """
+
+      expected_params = {
+        'account_id': '12001',
+        'project_id': '111001',
+        'visitors': [{
+          'visitor_id': 'test_user',
+          'attributes': [],
+          'snapshots': [{
+            'decisions': [{
+              'variation_id': '111129',
+              'experiment_id': '111127',
+              'campaign_id': '111182'
+            }],
+            'events': [{
+              'timestamp': 42123,
+              'entity_id': '111182',
+              'uuid': 'a68cf1ad-0393-4e18-af87-efe8f01a7c9c',
+              'key': 'campaign_activated'
+            }]
+          }]
+        }],
+        'client_name': 'python-sdk',
+        'client_version': version.__version__,
+        'anonymize_ip': False
+      }
+
+      with mock.patch('time.time', return_value=42.123), \
+           mock.patch('uuid.uuid4', return_value='a68cf1ad-0393-4e18-af87-efe8f01a7c9c'):
+        event_obj = self.event_builder.create_impression_event(
+          self.project_config.get_experiment_from_key('test_experiment'),
+          '111129', 'test_user', {'do_you_know_me': 'test_value'}
+        )
+      self._validate_event_object(event_obj,
+                                  event_builder.EventBuilder.EVENTS_URL,
+                                  expected_params,
+                                  event_builder.EventBuilder.HTTP_VERB,
+                                  event_builder.EventBuilder.HTTP_HEADERS)
+
+  def test_create_conversion_event(self):
```

Comment:
Why are these tests being added?

Reply:
I found these two scenarios missing, which we have covered in other SDKs. 
1. test that expected impression payload is created when an attribute, which is not in the datafile, is given. 
2. test that expected conversion event is created when no attributes are passed at all.